### PR TITLE
chore: debug APIC configuration and readme docs

### PR DESCRIPTION
**Pull request by leoporoli**
## Description:
debug APIC configuration and readme docs

## Is this change user facing?
NO

### CODE REVIEW

File: cli/cli/commands/run/run.go [hunk 1]
```diff
@@ -562,9 +563,9 @@ func getOrCreateEnclaveContext(
 	var enclaveContext *enclaves.EnclaveContext
 	var err error
 	if isProduction {
-		enclaveContext, err = kurtosisContext.CreateProductionEnclave(ctx, enclaveIdentifierOrName)
+		enclaveContext, err = kurtosisContext.CreateProductionEnclave(ctx, enclaveIdentifierOrName, defaults.DefaultEnableDebugMode)
 	} else {
-		enclaveContext, err = kurtosisContext.CreateEnclave(ctx, enclaveIdentifierOrName)
+		enclaveContext, err = kurtosisContext.CreateEnclave(ctx, enclaveIdentifierOrName, defaults.DefaultEnableDebugMode)
```
Comment (h4ck3rk3y) [inline cli/cli/commands/run/run.go]: does this mean anyone using the SDK has to now pass this? Making this a breaking change.

I wonder if we can make it CreateEnclaveWithDebugEnabled and  CreateProductionEnclaveWithDebugenabled; can do this later too. Maybe we can add a `EnclaveCreationOptions` or something.

File: core/scripts/build.sh [hunk 2]
```diff
@@ -10,19 +10,41 @@ root_dirpath="$(dirname "${script_dirpath}")"
...
     "files_artifacts_expander/scripts/build.sh"
 )
 
+# ==================================================================================================
+#                                       Arg Parsing & Validation
+# ==================================================================================================
+show_helptext_and_exit() {
+    echo "Usage: $(basename "${0}") debug_image..."
+    echo ""
+    echo "  debug_image   Whether images should contains the debug server and run in debug mode, this will use the Dockerfile.debug image to build the container (only configured for the engine server so far)"
```
Comment (h4ck3rk3y) [inline core/scripts/build.sh]: does the `only configured for engine server so far` apply?

File: scripts/port-forward-apic-debug.sh [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+
+# ==================================================================================================
+#                                             Main Logic
+# ==================================================================================================
+apic_namespace_name="${KURTOSIS_KUBERNETES_NAMESPACE_PREFIX}-${enclave_name}"
+
+get_apic_pod_name_cmd="kubectl get pods -l="${APIC_CONTAINER_K8S_LABEL}" -n ${apic_namespace_name} | cut -d ' ' -f 1  | tail -n +2"
+
+# execute the port forward to the debug server inside the engine's container
```
Comment (h4ck3rk3y) [inline scripts/port-forward-apic-debug.sh]: do you mean apics container?

**Review by h4ck3rk3y — APPROVED** [review]

File: core/scripts/build.sh (hunk 2 above)
Comment (leoporoli) [reply core/scripts/build.sh]: No, I doesn't apply, fixing it...

File: scripts/port-forward-apic-debug.sh (hunk 3 above)
Comment (leoporoli) [reply scripts/port-forward-apic-debug.sh]: yes, I do, thanks, fixing it...

**Pull request merged**